### ISSUE TITLE
"Empty the Trash" deletes all files without any sort of warning.

### ISSUE THREAD

**Issue by Blast-City** [elementary/files#1719]
## What Happened

As soon as the "Empty the Trash" button is pressed the files are deleted without any sort of warning/confirmation that this will be permanent.

## Expected Behavior

Clicking "Empty the Trash" should warn the user that this operation is irreversible.

## Steps to Reproduce

1. Create a file
2. Right click on the file and select "Move to Trash"
3. Go to Trash and click on "Empty the Trash"

## Platform Information

elementary OS 5.1.7 Hera

**Comment by jeremypw** [comment]
I am not sure we need a confirmation dialog but the text should be e.g. "Permanently delete all trash" maybe with destructive action styling.

**Comment by Blast-City** [comment]
I disagree. Accidents happen, especially with "tap to click" enabled so it's always good to have a safe net for that cases.

What are the advantages of not having confirmation dialog?

**Comment by danrabbit** [comment]
The entire trash is a bit of an undo/confirm flow isn't it? If you've moved something to the trash, why wouldn't you expect it to be deleted?

**Comment by jeremypw** [comment]
I don't think there is a way of permanently deleting anything with only one action (or there shouldn't be).  The second action should always be preceded by some indication that the operation is irreversible.  Sometimes it is a dialog (e.g. after using the shortcut) but otherwise it is text/color on the button/menuitem. A third required action seems excessive.

**Comment by jeremypw** [comment]
May I ask whether you have actually lost files through accidentally deleting them from the trash?

**Comment by Blast-City** [comment]
> The entire trash is a bit of an undo/confirm flow isn't it? If you've moved something to the trash, why wouldn't you expect it to be deleted?

That makes sense, but since we're talking about an irreversible action I don't see why not just give an extra warning. Empty the Trash is something that people use sporadically so it's not that bothersome to have to do an extra click.

> May I ask whether you have actually lost files through accidentally deleting them from the trash?

No, but I was testing something and then I moved the test files to Trash. When I went to empty the Trash I was really surprised that it permanently removed everything without any confirmation. I'm used to see a confirmation on other Apps so it felt like something was off.

**Comment by Blast-City** [comment]
Another issue that can occur it when you're trying to restore one file, the "restore form Trash" context menu entry is so close to the "delete permanently" that it takes just a twitch and your file is removed forever.

**Comment by jeremypw** [comment]
Yes, that's probably a valid issue. I would tend to just separate those two items and style it as a destructive action but I guess "Delete permanently" is always going to be next to something.

**Comment by jeremypw** [comment]
Another solution would be to remove the "Delete Permanently" from the context menu and require the user to open a trash view and use the buttons, which are larger targets.  Needs UX team input.
